File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> hash;
        
        for(int itr : nums) {
            if(hash.find(itr) != hash.end())
                hash[itr]++;
            else
                hash.insert({itr, 1});
        }
        
        unordered_map<int, int>::iterator itr;
        for(itr = hash.begin(); itr != hash.end(); itr++) {
            if(itr->second > nums.size()/2) return itr->first;
        }
        
        return -1;
    }
};

File: 169-majority-element/169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let curEl, count = 0;
    
    for(let i = 0; i < nums.length; i++) {
        if(count == 0) {
            curEl = nums[i];
            count++;
        } else {
            if(curEl == nums[i]) count++;
            else count--;
        }
    }
    
    console.log(count);
    
    return curEl;
};